### PR TITLE
Online banking Poland

### PR DESCRIPTION
**Pull request by etman55**
## Summary
Add online banking payment method for Poland

### CODE REVIEW

File: drop-in/src/main/java/com/adyen/checkout/dropin/DropInConfiguration.kt [hunk 1]
```diff
@@ -270,6 +271,11 @@ class DropInConfiguration : Configuration, Parcelable {
             return this
         }
 
+        fun addOnlineBankingPLConfiguration(onlineBankingPLConfiguration: OnlineBankingPLConfiguration): Builder {
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/DropInConfiguration.kt:274]: Minor: add method docs

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 2]
```diff
@@ -140,6 +143,9 @@ internal fun <T : Configuration> getDefaultConfigForPaymentMethod(
         PaymentMethodTypes.BCMC -> BcmcConfiguration.Builder(shopperLocale, environment, clientKey)
         PaymentMethodTypes.BLIK -> BlikConfiguration.Builder(shopperLocale, environment, clientKey)
         PaymentMethodTypes.DOTPAY -> DotpayConfiguration.Builder(shopperLocale, environment, clientKey)
+        PaymentMethodTypes.ONLINE_BANKING_PL -> OnlineBankingPLConfiguration.Builder(
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:146]: Nit: the `PaymentMethodTypes` in this `when` are sorted alphabetically 😅

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 3]
```diff
@@ -311,6 +317,11 @@ internal fun getComponentFor(
                 getConfigurationForPaymentMethod(PaymentMethodTypes.DOTPAY, dropInConfiguration, amount)
             DotpayComponent.PROVIDER.get(fragment, paymentMethod, dotpayConfig)
         }
+        PaymentMethodTypes.ONLINE_BANKING_PL -> {
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:320]: Same, alphabetical order

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 4]
```diff
@@ -403,6 +414,7 @@ internal fun getViewFor(
     return when (paymentType) {
         PaymentMethodTypes.BCMC -> BcmcView(context)
         PaymentMethodTypes.DOTPAY -> DotpayRecyclerView(context)
+        PaymentMethodTypes.ONLINE_BANKING_PL -> OnlineBankingPLRecyclerView(context)
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:417]: Same

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 5]
```diff
@@ -503,6 +515,7 @@ private fun Configuration.toBuilder(): BaseConfigurationBuilder<out Configuratio
         is BlikConfiguration -> BlikConfiguration.Builder(this)
         is CardConfiguration -> CardConfiguration.Builder(this)
         is DotpayConfiguration -> DotpayConfiguration.Builder(this)
+        is OnlineBankingPLConfiguration -> OnlineBankingPLConfiguration.Builder(this)
```
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:518]: Same (really sorry 😂)

File: example-app/src/main/java/com/adyen/checkout/example/data/api/model/PaymentsRequest.kt [hunk 6]
```diff
@@ -32,5 +32,6 @@ data class PaymentsRequestData(
     val reference: String,
     val channel: String,
     val lineItems: List<Item>,
+    val shopperEmail: String? = null,
```
Comment (jreij) [inline example-app/src/main/java/com/adyen/checkout/example/data/api/model/PaymentsRequest.kt:35]: Can you test this with sessions? Maybe we need to add this field to the `SessionRequest` as well

File: onlinebanking/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (c) 2022 Adyen N.V.
+  ~
+  ~ This file is open source and available under the MIT license. See the LICENSE file for more info.
+  ~
+  ~ Created by atef on 9/8/2022.
+  -->
+
+<manifest package="com.adyen.checkout.onlinebanking">
```
Comment (jreij) [inline onlinebanking/src/main/AndroidManifest.xml:10]: Can we use this package name? If `PL` is different than the other components then we probably need to change the package name as well

File: onlinebanking/src/test/java/com/adyen/checkout/onlinebanking/ExampleUnitTest.kt [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import org.junit.Test
+
+import org.junit.Assert.*
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
```
Comment (jreij) [inline onlinebanking/src/test/java/com/adyen/checkout/onlinebanking/ExampleUnitTest.kt:20]: Let's remove this file?

File: settings.gradle [hunk 9]
```diff
@@ -26,5 +26,5 @@ include ':3ds2',
         ':test-core',
         ':ui-core',
         ':voucher',
-        ':wechatpay'
-
+        ':wechatpay',
+        ':onlinebanking'
```
Comment (jreij) [inline settings.gradle:30]: Alphabetical order here as well

File: components-core/src/main/java/com/adyen/checkout/components/model/payments/request/OnlineBankingPLPaymentMethods.kt [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+import android.os.Parcel
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.core.exception.ModelSerializationException
+import com.adyen.checkout.core.model.JsonUtils
+import com.adyen.checkout.core.model.getStringOrNull
+import org.json.JSONException
+import org.json.JSONObject
+
+class OnlineBankingPLPaymentMethods(
```
Comment (OscarSpruit) [inline components-core/src/main/java/com/adyen/checkout/components/model/payments/request/OnlineBankingPLPaymentMethods.kt:19]: Shouldn't this be `OnlineBankingPLPaymentMethod` (minus the "s")?

File: onlinebanking/src/main/java/com/adyen/checkout/onlinebanking/OnlineBankingPLRecyclerView.kt [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+package com.adyen.checkout.onlinebanking
+
+import android.content.Context
+import android.util.AttributeSet
+import com.adyen.checkout.components.model.payments.request.OnlineBankingPLPaymentMethods
+import com.adyen.checkout.issuerlist.IssuerListRecyclerView
+
+class OnlineBankingPLRecyclerView : IssuerListRecyclerView<OnlineBankingPLPaymentMethods, OnlineBankingPLComponent> {
+    constructor(context: Context) : super(context)
```
Comment (OscarSpruit) [inline onlinebanking/src/main/java/com/adyen/checkout/onlinebanking/OnlineBankingPLRecyclerView.kt:17]: `@JvmOverloads` can also be used

File: onlinebanking/src/main/java/com/adyen/checkout/onlinebanking/OnlineBankingPLConfiguration.kt [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+import android.content.Context
+import android.os.Parcel
+import android.os.Parcelable
+import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.issuerlist.IssuerListConfiguration
+import java.util.Locale
+
+class OnlineBankingPLConfiguration : IssuerListConfiguration {
+
+    internal constructor(
```
Comment (OscarSpruit) [inline onlinebanking/src/main/java/com/adyen/checkout/onlinebanking/OnlineBankingPLConfiguration.kt:20]: Shouldn't these constructors be `private`?

File: example-app/src/main/java/com/adyen/checkout/example/data/api/model/PaymentsRequest.kt (hunk 6 above)
Comment (etman55) [reply example-app/src/main/java/com/adyen/checkout/example/data/api/model/PaymentsRequest.kt:35]: yes it's not working with session I'll add it in `SessionRequest`

File: onlinebanking/src/main/java/com/adyen/checkout/onlinebanking/OnlineBankingPLConfiguration.kt (hunk 12 above)
Comment (etman55) [reply onlinebanking/src/main/java/com/adyen/checkout/onlinebanking/OnlineBankingPLConfiguration.kt:20]: Yes, you're right I just copied and pasted it from `IdealConfiguration`

**Review by jreij — APPROVED** [review]

**Pull request merged**